ibis/common/selectors.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+import abc
+from typing import TYPE_CHECKING
+
+from ibis.common.grounds import Concrete
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    import ibis.expr.types as ir
```

Comment:
I would prefer not to move the base selector class to `ibis.common` since it is tied to `ir.Table`. 
I think `Selector`s could be viewed as another kind of deferred producing a sequence of outputs and deferreds are decoupled from the `ir`.

Reply:
If I recall correctly even `bind()` would properly handle that, so refactoring selectors as `Resolver`s could simplify the codebase, but that should be done in a separate PR.

Reply:
I moved the class to avoid an import cycle and having it anywhere else didn't really fit.

If you have another suggestion that isn't to refactor the whole class :) I'm happy to hear it!

Reply:
Ohh, I wasn't aware of the import cycle.